kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/EndPoint.java

```diff
@@ -31,26 +31,45 @@
 public class EndPoint {
 
     private static final String END_POINT_SEPARATOR = ",";
+    private static final String PROTOCOL_MAP_SEPARATOR = ",";
+    private static final String PROTOCOL_SEPARATOR = ":";
     private static final String REGEX = "^(.*)://\\[?([0-9a-zA-Z\\-%._:]*)\\]?:(-?[0-9]+)";
     private static final Pattern PATTERN = Pattern.compile(REGEX);
 
     @Getter
     private final String originalListener;
     @Getter
+    private final String listenerName;
+    @Getter
     private final SecurityProtocol securityProtocol;
     @Getter
     private final String hostname;
     @Getter
     private final int port;
+    @Getter
+    private final boolean multiListener;
 
-    public EndPoint(final String listener) {
+    public EndPoint(final String listener, final String kafkaProtocolMap) {
         this.originalListener = listener;
         final String errorMessage = "listener '" + listener + "' is invalid";
         final Matcher matcher = PATTERN.matcher(listener);
         checkState(matcher.find(), errorMessage);
         checkState(matcher.groupCount() == 3, errorMessage);
 
-        this.securityProtocol = SecurityProtocol.forName(matcher.group(1));
+        Map<String, SecurityProtocol> protocolMap = parseProtocolMap(kafkaProtocolMap);
```

Comment:
We can pass a `Map<String, SecurityProtocol>` to the constructor's parameter to avoid calling `parseProtocolMap` every time a `EndPoint` is constructed.

Reply:
done